patroni/api.py

```diff
@@ -89,8 +90,17 @@ def do_GET(self, write_status_code_only=False):
         patroni = self.server.patroni
         cluster = patroni.dcs.cluster
 
-        replica_status_code = 200 if not patroni.noloadbalance and \
-            response.get('role') == 'replica' and response.get('state') == 'running' else 503
+        replica_status_code = 503
+        if response.get('role') == 'replica' and response.get('state') == 'running' \
+           and not patroni.noloadbalance:
+            max_replica_lag = self.path_query.get('lag') and parse_int(self.path_query.get('lag')[0])
+            leader_optime = parse_int(patroni.dcs.leader_optime)
```

Comment:
We should really avoid calling the DCS from REST API.
There is already `cluster.last_leader_operation` that contains the value from `/optime/leader`, the only minor problem that this value sometimes is outdated, but https://github.com/zalando/patroni/pull/1600 solves it.

Reply:
Agree. I have tried initially with cluster.last_leader_operation and also cluster.leader.member.data.xlog_location. Fixing last_leader_operation greatly simplifies the effort and improves performance. We can fix the code to use cluster.last_leader_operation. Thanks.